localstack-core/localstack/services/lambda_/resource_providers/aws_lambda_eventsourcemapping.py

```diff
@@ -126,8 +127,17 @@ def create(
         model = request.desired_state
         lambda_client = request.aws_client_factory.lambda_
 
-        response = lambda_client.create_event_source_mapping(**model)
+        params = copy.deepcopy(model)
+        if tags := params.get("Tags"):
+            transformed_tags = {}
+            for tag_definition in tags:
+                transformed_tags[tag_definition["Key"]] = tag_definition["Value"]
+            params["Tags"] = transformed_tags
+
+        response = lambda_client.create_event_source_mapping(**params)
         model["Id"] = response["UUID"]
+        model["EventSourceArn"] = response["EventSourceArn"]
```

Comment:
Just want to clarify that this value needs to be set? I see the CFn docs show us just being able to get the `Id` and `EventSourceMappingArn` attributes.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-eventsourcemapping.html#aws-resource-lambda-eventsourcemapping-return-values

Reply:
You are right, thanks 🙏 . I have just tested against AWS (never trust the docs, _especially_ for CFn!). On the other hand, there is a discrepancy between the specs we use to generate the resource providers and the CFn implementation. The specs do not mention `EventSourceMappingArn` as a property that can be accessed on the resource, however it is clearly there in the docs.

Again: _never_ **_NEVER_** trust the docs! (or the specs in this case... 😂 )